test/catch2_test_helper.h

```diff
@@ -0,0 +1,253 @@
+/******************************************************************************
+* Copyright (c) 2011-2021, NVIDIA CORPORATION.  All rights reserved.
+*
+* Redistribution and use in source and binary forms, with or without
+* modification, are permitted provided that the following conditions are met:
+*     * Redistributions of source code must retain the above copyright
+*       notice, this list of conditions and the following disclaimer.
+*     * Redistributions in binary form must reproduce the above copyright
+*       notice, this list of conditions and the following disclaimer in the
+*       documentation and/or other materials provided with the distribution.
+*     * Neither the name of the NVIDIA CORPORATION nor the
+*       names of its contributors may be used to endorse or promote products
+*       derived from this software without specific prior written permission.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+* ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+* WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+* DISCLAIMED. IN NO EVENT SHALL NVIDIA CORPORATION BE LIABLE FOR ANY
+* DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+* (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+* LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+* ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+* (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+* SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*
+******************************************************************************/
+
+#pragma once
+
+#include <thrust/device_vector.h>
+#include <thrust/distance.h>
+#include <thrust/tabulate.h>
+
+#include <type_traits>
+#include <curand.h>
+#include <cstdint>
+#include <tuple>
+
+#include "cub/util_compiler.cuh"
+
+// Workaround for unused variables in catch2
+#if CUB_HOST_COMPILER == CUB_HOST_COMPILER_MSVC
+#pragma diag_suppress 177
+#elif CUB_HOST_COMPILER == CUB_HOST_COMPILER_CLANG
+#pragma clang diagnostic ignored "-Wunused-variable"
+#endif
+
+#include <catch2/catch.hpp>
+
+// It's a port of cartesian product from https://github.com/NVIDIA/nvbench
+namespace catch2_helper
```

Comment:
I'd love if we could contribute this to Catch2. I filed an issue for exactly this a long time ago https://github.com/catchorg/Catch2/issues/1989